compiler/rustc_session/src/lint/builtin.rs

```diff
@@ -2555,6 +2555,13 @@ declare_lint! {
     @feature_gate = sym::unsafe_block_in_unsafe_fn;
 }
 
+declare_lint! {
+    //TODO: Add explanation.
+    pub FALL_BACK_TO_NEVER_TYPE,
+    Deny,
+    "Unresolved variable might fall back to never_type `!`"
```

Comment:
```suggestion
    "Unresolved variable might fall back to never_type `!` in the future"
```

Should this be a future compatibility lint?

Reply:
Probably, but we're still tinkering with getting it to issue warnings at the right times. That said, I think that this lint may make sense even once the transition has happened, as it represents a surprising situation most of the time.